File: examples/nodejs/generate-credential.js
/**
 * Generate NIST-compliant credentials using the Six Sense API
 * 
 * Get your free API key at sixsensesolutions.net
 */

const API_KEY = process.env.SIX_SENSE_API_KEY;
const BASE_URL = 'https://api.sixsensesolutions.net';

async function generateNISTCredential() {
  const response = await fetch(`${BASE_URL}/v1/generate`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      length: 20,
      quantity: 1,
      compliance: 'NIST',
      options: {
        uppercase: true,
        lowercase: true,
        numbers: true,
        symbols: true,
        exclude_ambiguous: true
      }
    })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API error: ${error.error} - ${error.message}`);
  }

  const { passwords, meta } = await response.json();

  console.log('Generated credential (store securely, not logged here)');
  console.log('Compliance documentation:');
  console.log(`  Length: ${meta.length}`);
  console.log(`  Entropy: ${meta.entropy_bits} bits`);
  console.log(`  Profile: ${meta.compliance_profile}`);
  console.log(`  Generated at: ${meta.generated_at}`);
  console.log(`  Calls remaining: ${meta.calls_remaining}`);

  return { credential: passwords[0], meta };
}

generateNISTCredential().catch(console.error);

File: examples/python/generate_credential.py
"""
Generate NIST-compliant credentials using the Six Sense API

Get your free API key at sixsensesolutions.net
"""

import os
import requests

API_KEY = os.environ.get('SIX_SENSE_API_KEY')
BASE_URL = 'https://api.sixsensesolutions.net'


def generate_nist_credential(length=20, quantity=1):
    response = requests.post(
        f'{BASE_URL}/v1/generate',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={
            'length': length,
            'quantity': quantity,
            'compliance': 'NIST',
            'options': {
                'uppercase': True,
                'lowercase': True,
                'numbers': True,
                'symbols': True,
                'exclude_ambiguous': True
            }
        },
        timeout=15
    )

    response.raise_for_status()
    data = response.json()

    meta = data['meta']
    print('Compliance documentation:')
    print(f"  Length: {meta['length']}")
    print(f"  Entropy: {meta['entropy_bits']} bits")
    print(f"  Profile: {meta['compliance_profile']}")
    print(f"  Generated at: {meta['generated_at']}")
    print(f"  Calls remaining: {meta['calls_remaining']}")

    return data['passwords'], meta


if __name__ == '__main__':
    passwords, meta = generate_nist_credential()
